spec/controllers/catalog_controller_spec.rb

```diff
@@ -54,6 +54,51 @@
       end
     end
   end
+  describe 'POST submit_form' do
```

Comment:
You need to also test for the final result of each request. In this case, you're redirecting to the new reservation page, so let's verify that that happens.